### ISSUE TITLE
[meta2] In rare case we encounter multiple meta2 master 

### ISSUE THREAD

**Issue by sebastienlapierre** [open-io/oio-sds#1336]
openio-sds: 4.1.11-1

Here is election status and logs during a object PUT.
Election status:
```
{
    "127.0.0.1:6121": {
        "body": {
            "#": {
                "P": 0,
                "R": 364,
                "V": 0
            },
            "base": {
                "name": "B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1",
                "type": "meta2",
                "zk": "3B84DBD6E01AB2158ECEB41F77B60FFBEC38AF4330C9E86DDC347146F6E6931C"
            },
            "local": {
                "id": 798,
                "state": "SLAVE",
                "url": "127.0.0.1:6121"
            },
            "log": [
                "CHECKING_MASTER:GETVERS_OK:SLAVE",
                "ASKING:MASTER_OK:CHECKING_MASTER",
                "LISTING:LIST_OK:ASKING",
                "WATCHING:EXISTS_OK:LISTING",
                "CREATING:CREATE_OK:WATCHING",
                "NONE:NONE:CREATING"
            ],
            "master": {
                "id": 708,
                "url": "127.0.0.2:6121"
            },
            "peers": [
                "127.0.0.3:6121",
                "127.0.0.2:6121"
            ]
        },
        "status": {
            "message": "OK",
            "status": 200
        }
    },
    "127.0.0.2:6121": {
        "body": {
            "#": {
                "P": 0,
                "R": 262,
                "V": 0
            },
            "base": {
                "name": "B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1",
                "type": "meta2",
                "zk": "3B84DBD6E01AB2158ECEB41F77B60FFBEC38AF4330C9E86DDC347146F6E6931C"
            },
            "local": {
                "id": 708,
                "state": "MASTER",
                "url": "127.0.0.2:6121"
            },
            "log": [
                "CHECKING_SLAVES:GETVERS_OK:MASTER",
                "CHECKING_SLAVES:GETVERS_OK:CHECKING_SLAVES",
                "LISTING:LIST_OK:CHECKING_SLAVES",
                "WATCHING:EXISTS_OK:LISTING",
                "CREATING:CREATE_OK:WATCHING",
                "NONE:NONE:CREATING",
                "LEAVING:LEAVE_OK:NONE",
                "CHECKING_MASTER:GETVERS_OK:SLAVE",
                "ASKING:MASTER_OK:CHECKING_MASTER",
                "LISTING:LIST_OK:ASKING",
                "WATCHING:EXISTS_OK:LISTING",
                "CREATING:CREATE_OK:WATCHING",
                "NONE:NONE:CREATING",
                "LEAVING:LEAVE_OK:NONE",
                "LEAVING:LEFT_SELF:LEAVING",
                "CHECKING_MASTER:GETVERS_OK:SLAVE"
            ],
            "master": {
                "id": 708,
                "url": null
            },
            "peers": [
                "127.0.0.3:6121",
                "127.0.0.1:6121"
            ]
        },
        "status": {
            "message": "OK",
            "status": 200
        }
    },
    "127.0.0.3:6121": {
        "body": {
            "#": {
                "P": 0,
                "R": 6,
                "V": 0
            },
            "base": {
                "name": "B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1",
                "type": "meta2",
                "zk": "3B84DBD6E01AB2158ECEB41F77B60FFBEC38AF4330C9E86DDC347146F6E6931C"
            },
            "local": {
                "id": 177,
                "state": "MASTER",
                "url": "127.0.0.3:6121"
            },
            "log": [
                "CHECKING_SLAVES:GETVERS_OK:MASTER",
                "CHECKING_SLAVES:GETVERS_KO:CHECKING_SLAVES",
                "LISTING:LIST_OK:CHECKING_SLAVES",
                "SLAVE:LEFT_MASTER:LISTING",
                "CHECKING_MASTER:GETVERS_OK:SLAVE",
                "ASKING:MASTER_OK:CHECKING_MASTER",
                "LISTING:LIST_OK:ASKING",
                "SLAVE:LEFT_MASTER:LISTING",
                "CHECKING_MASTER:GETVERS_OK:SLAVE",
                "ASKING:MASTER_OK:CHECKING_MASTER",
                "LISTING:LIST_OK:ASKING",
                "WATCHING:EXISTS_OK:LISTING",
                "CREATING:CREATE_OK:WATCHING",
                "NONE:NONE:CREATING"
            ],
            "master": {
                "id": 177,
                "url": null
            },
            "peers": [
                "127.0.0.1:6121",
                "127.0.0.2:6121"
            ]
        },
        "status": {
            "message": "OK",
            "status": 200
        }
    }
}
```
oioswift+oioproxy logs:
```
/var/log/oio/sds/OPENIO/oioproxy-1/oioproxy-1.access:2018-01-23T12:17:43.621792+00:00 127.0.0.4 OIO,OPENIO,oioproxy,1: info  11024 C5B5 access INF 127.0.0.4:6006 127.0.0.4:19702 POST 200 5674076 611 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 /v3.0/OPENIO/container/get_properties?acct=ACCOUNT&ref=container t=5674022
/var/log/oio/sds/OPENIO/oioproxy-1/oioproxy-1.access:2018-01-23T12:17:43.626724+00:00 127.0.0.4 OIO,OPENIO,oioproxy,1: info  11024 9291 access INF 127.0.0.4:6006 127.0.0.4:20856 POST 200 1161 1098 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 /v3.0/OPENIO/content/prepare?acct=ACCOUNT&ref=container&path=object1 t=1090
/var/log/oio/sds/OPENIO/oioproxy-1/oioproxy-1.access:2018-01-23T12:17:53.963376+00:00 127.0.0.4 OIO,OPENIO,oioproxy,1: info  11024 CB95 access INF 127.0.0.4:6006 127.0.0.4:20872 POST 503 10328676 164 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 /v3.0/OPENIO/content/create?acct=ACCOUNT&ref=container&path=object1 t=10328591  e={"status":503,"message":"COMMIT failed: (SQLITE_CONSTRAINT) constraint failed [127.0.0.1:6121\/472\/Concurrent change detected] [127.0.0.2:6121\/7\/Timeout]"}

/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:37.928208+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: info  - - 23/Jan/2018/12/17/37 HEAD /container HTTP/1.0 400 - Swift - - - - tx1111-2222 - 0.0003 RL - 1516709857.927630901 1516709857.927894115 -
/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:37.929616+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: info  - - 23/Jan/2018/12/17/37 HEAD /container HTTP/1.0 400 - Swift - - - - tx1111-2222 - 0.0002 RL - 1516709857.929240942 1516709857.929431915 -
/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:43.623078+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: info  - - 23/Jan/2018/12/17/43 HEAD /v1/ACCOUNT/container HTTP/1.0 204 - Swift - - - - tx1111-2222 - 5.6759 CQ - 1516709857.946937084 1516709863.622807026 -
/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:53.964749+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: info  127.0.0.12 127.0.0.12 23/Jan/2018/12/17/53 PUT /v1/ACCOUNT/container/object1 HTTP/1.0 503 - fog/1.41.0%20fog-core/1.45.0 - 1195 133 - tx1111-2222 - 16.0182 S3 - 1516709857.946307898 1516709873.964463949 -
/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:53.965151+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: err  500 Internal Server Error: #012Traceback (most recent call last):#012  File "/usr/lib/python2.7/dist-packages/swift3/middleware.py", line 119, in __call__#012    resp = self.handle_request(req)#012  File "/usr/lib/python2.7/dist-packages/swift3/middleware.py", line 150, in handle_request#012    res = handler(req)#012  File "/usr/lib/python2.7/dist-packages/swift3/controllers/obj.py", line 110, in PUT#012    resp = req.get_response(self.app)#012  File "/usr/lib/python2.7/dist-packages/swift3/request.py", line 1370, in get_response#012    headers, body, query)#012  File "/usr/lib/python2.7/dist-packages/swift3/request.py", line 1355, in _get_response#012    raise InternalError('unexpected status code %d' % status)#012InternalError: 500 Internal Server Error (txn: tx1111-2222) (client_ip: 127.0.0.12)
/var/log/oio/sds/OPENIO/oioswift-1/oioswift-1.log:2018-01-23T12:17:53.965767+00:00 127.0.0.4 OIO,OPENIO,oioswift,1: info  127.0.0.12 127.0.0.12 23/Jan/2018/12/17/53 PUT /container/object1 HTTP/1.0 500 - fog/1.41.0%20fog-core/1.45.0 - 1195 182 - tx1111-2222 - 16.0388 - - 1516709857.926801920 1516709873.965557098 - 
```

META2 LOGS
127.0.0.1
``` 
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:43.624454+00:00 openio01 OIO,OPENIO,meta2,1: info  23719 B893 access INF 127.0.0.1:6121 127.0.0.4:21210 M2_PREP 200 184 1220 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=161 OPENIO/ACCOUNT/container/object1|B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:43.638420+00:00 openio01 OIO,OPENIO,meta2,1: info  23719 B893 access INF 127.0.0.1:6121 127.0.0.3:20408 DB_REPLI 472 1958 93 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=1880 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1.meta2 e=(472) Concurrent change detected
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:53.960976+00:00 openio01 OIO,OPENIO,meta2,1: info  23719 ADDB access INF 127.0.0.1:6121 127.0.0.3:23184 DB_RESTORE 200 67793 69 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=62600 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1.meta2
```

127.0.0.2
``` 
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:53.784989+00:00 openio02 OIO,OPENIO,meta2,1: info  39603 9B5E access INF 127.0.0.2:6121 127.0.0.3:58076 DB_REPLI 200 10143846 69 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=10143798 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1.meta2 
``` 

127.0.0.3
``` 
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:43.620342+00:00 127.0.0.3 OIO,OPENIO,meta2,1: info  15979 5F5C access INF 127.0.0.3:6121 127.0.0.4:49948 DB_PGET 200 5673142 659 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=5673116 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A.1.meta2
/var/log/oio/sds/OPENIO/meta2-1/meta2-1.access:2018-01-23T12:17:53.962223+00:00 127.0.0.3 OIO,OPENIO,meta2,1: info  15979 6491 access INF 127.0.0.3:6121 127.0.0.4:51120 M2_PUT 503 10327531 204 B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A tx1111-2222 t=10327492 OPENIO/ACCOUNT/container/object1?id=550EA688706305009ECF0B3FA75C93B8|B38CCCB618276081320BAC72B0A59D8E6D716255598092D312D7815A9451A30A e=(503) COMMIT failed: (SQLITE_CONSTRAINT) constraint failed [127.0.0.1:6121/472/Concurrent change detected] [127.0.0.2:6121/7/Timeout]
```

**Comment by jkasarherou** [comment]
Got same issue using `4.1.12`. Any updates on this?

**Comment by jfsmig** [comment]
Fixed in Release 4.2.6 (and subsequent in the 4.2.x branch), and propagated in 4.3.1 (and subsequent in 4.3.x branch)
